jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/AbstractUrlResourceFactory.java

```diff
@@ -0,0 +1,234 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util.resource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Path;
+import java.time.Instant;
+
+import org.eclipse.jetty.util.FileID;
+
+/**
+ * Abstract ResourceFactory for {@link java.net.URL} based resources.
+ */
+public abstract class AbstractUrlResourceFactory implements ResourceFactory
+{
+    private final String supportedProtocol;
```

Comment:
This class does not need to be abstract.  It just has the protocol injected via the constructor and it creates a needless amount of work to subclass just to do that.

Actually, it doesn't even need to check the protocol, as the factory is only registered for the protocols that we wish it to handle, thus a second check is duplication.

At the very least, it should have a set of protocols, as we don't want to create a new instance for every protocol if we have multiple, but I would prefer no check and just let the registration map do it's job.

Reply:
I'm leaving the check there.

The protocol will be used ....

* to ensure that an old `ResourceFactory` instance isn't being reused for a different protocol.
* for output on `toString()` to indicate the supported protocol.
* for configuring the URLConnection via protocol based system properties (usecaches, connecttimeout, etc).

There's no harm in having multiple `UrlResourceFactory` implementations, one for each requested protocol, it's a support layer for old school behaviors anyway.